### PR TITLE
Example Knative PowerShell Email Function

### PR DESCRIPTION
**Pull request by lamw**
Signed-off-by: William Lam <wlam@vmware.com>

## Summary

An Example Knative PowerShell function that uses [Send-MailKitMessage](https://www.powershellgallery.com/packages/Send-MailKitMessage) to send email notification.

screenshot-1

## Pull Request Checklist
🚨 Please review the [guidelines for contributing](https://vmweventbroker.io/community) to this repository.

- [ ] Please ensure that you are making a pull request against the **Development** branch
- [ ] Please use the `WIP` keyword in the title of your PR if you are not ready for review
- [ ] Please ensure that you have opened a Github Issue if you are resolving/fixing a problem
- [ ] Please ensure that you have [signed](https://help.github.com/en/github/authenticating-to-github/signing-commits) all commits and that you have [squashed](https://medium.com/@slamflipstrom/a-beginners-guide-to-squashing-commits-with-git-rebase-8185cf6e62ec) all relevant commits related to your change
- [ ] Please make sure that you have tested your change locally by successfully [building and deploying the VMware Event Broker Appliance](https://vmweventbroker.io/kb/contribute-appliance) and/or [building and deploying VMware Event Router](https://vmweventbroker.io/kb/contribute-eventrouter)
- [ ] Please include any relevant screenshots and/or output as part of your testing
- [ ] Please include any documentation updates that is applicable for your changes

## Change Type

What types of changes does your code introduce to the VMware Event Broker Appliance?

_Put an `x` in all boxes that apply_

Please check the type of change your PR introduces:
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] Feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] Documentation changes
- [ ] Other (please describe):

## Resolved Issues

N/A

## Testing Verification

* Deployed function to VEBA

## Additional Information

N/A

### CODE REVIEW

File: examples/knative/powershell/kn-ps-email/README.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+```bash
+cd test
+```
+
+Update the following variable names  within the `docker-test-env-variable` file
+
+* SMTP_SERVER - Address of Email Server
+* SMTP_PORT - Port of Email Server
+* SMTP_USERNAME - Username to send email (use empty string for no username)
+* SMTP_PASSWORD - Password to send emaail (use empty string for no username)
```
Comment (embano1) [inline examples/knative/powershell/kn-ps-email/README.md:29]:
```suggestion
* SMTP_PASSWORD - Password to send email (use empty string for no password)
```

File: examples/knative/powershell/kn-ps-email/README.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+  "FullFormattedMessage": "Removed FooBar on 192.168.30.5 from Primp-Datacenter"
+}
+
+Sending Secure Email ...
+```
+
+# Step 3 - Deploy
+
+> **Note:** The following steps assume a working Knative environment using the
+`default` Rabbit broker. The Knative `service` and `trigger` will be installed in the
```
Comment (embano1) [inline examples/knative/powershell/kn-ps-email/README.md:99]:
```suggestion
`default` Rabbit `broker`. The Knative `service` and `trigger` will be installed in the
```

File: examples/knative/powershell/kn-ps-email/README.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+`vmware-functions` Kubernetes namespace, assuming that the broker is also available there.
```
Comment (embano1) [inline examples/knative/powershell/kn-ps-email/README.md:100]:
```suggestion
`vmware-functions` Kubernetes namespace, assuming that the `broker` is also available there.
```

File: examples/knative/powershell/kn-ps-email/README.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+Update the `email_secret.json` file with your email configurations and then create the kubernetes secret which can then accessed from within the function by using the environment variable named called `EMAIL_SECRET`.
```
Comment (embano1) [inline examples/knative/powershell/kn-ps-email/README.md:102]:
```suggestion
Update the `email_secret.json` file with your email configurations and then create the kubernetes secret which can then be accessed from within the function by using the environment variable named called `EMAIL_SECRET`.
```

File: examples/knative/powershell/kn-ps-email/test/send-cloudevent-test.sh [hunk 5]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+echo "Testing Function ..."
+curl -d@binary-payload -H "Content-Type: application/json" -H 'ce-specversion: 1.0' -H 'ce-id: id-123' -H 'ce-source: source-123' -H 'ce-type: binary' -H 'ce-subject: subject-123' -X POST localhost:8080
```
Comment (embano1) [inline examples/knative/powershell/kn-ps-email/test/send-cloudevent-test.sh:4]: IMHO better to use a real event type here 'ce-type: binary'

File: examples/knative/powershell/kn-ps-email/test/send-cloudevent-test.ps1 [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+$headers = @{
+    "Content-Type" = "application/json";
+    "ce-specversion" = "1.0";
+    "ce-id" = "id-123";
+    "ce-source" = "source-123";
+    "ce-type" = "binary";
+    "ce-subject" = "subject-123";
+}
+
+$body = Get-Content -Raw -Path "./binary-payload"
```
Comment (embano1) [inline examples/knative/powershell/kn-ps-email/test/send-cloudevent-test.ps1:11]: Should this be `test-payload.json` ?

File: examples/knative/powershell/kn-ps-email/test/send-cloudevent-test.sh (hunk 5 above)
Comment (embano1) [reply examples/knative/powershell/kn-ps-email/test/send-cloudevent-test.sh:4]: Where is `binary-payload` defined? Should this be `test-payload.json` ?

Comment (lamw) [reply examples/knative/powershell/kn-ps-email/test/send-cloudevent-test.sh:4]: What should the example be?

Comment (embano1) [reply examples/knative/powershell/kn-ps-email/test/send-cloudevent-test.sh]: I was just wondering and assuming that `-d` takes an existing file (name) but could not find `binary-payload` anywhere in the commit?

Comment (lamw) [reply examples/knative/powershell/kn-ps-email/test/send-cloudevent-test.sh]: Yup, this was my bad.

This is fixed w/latest commit - https://github.com/vmware-samples/vcenter-event-broker-appliance/blob/005a4e3308c423855d14ef7a7d1fed8460aee848/examples/knative/powershell/kn-ps-email/test/send-cloudevent-test.sh#L4

File: examples/knative/powershell/kn-ps-email/test/send-cloudevent-test.ps1 [hunk 7]
```diff
@@ -0,0 +1,16 @@
+
+$headers = @{
+    "Content-Type" = "application/json";
+    "ce-specversion" = "1.0";
+    "ce-id" = "id-123";
+    "ce-source" = "source-123";
+    "ce-type" = "binary";
+    "ce-subject" = "subject-123";
```
Comment (embano1) [inline examples/knative/powershell/kn-ps-email/test/send-cloudevent-test.ps1:8]: nit: leaving this note here for future changes: 

```suggestion
    "ce-source" = "source-123";
    "ce-type" = "com.vmware.event.router/event";
    "ce-subject" = "VmRemovedEvent";
```

**Review by embano1 — DISMISSED** [review]
LGTM

**Review by embano1 — APPROVED** [review]

**Pull request merged**